tests/integration/run_integration_tests.sh

```diff
@@ -0,0 +1,222 @@
+  NETWORK=$(docker container inspect kind-control-plane | jq -r '.[].NetworkSettings.Networks | to_entries | .[].key')
+fi
+
+echo "Spinning up mocked alerting interface..."
+docker run -d --name alerting-endpoint -p 56243:56243 docker.io/securesystemsengineering/alerting-endpoint:latest
+docker network connect ${NETWORK} alerting-endpoint
+export ALERTING_ENDPOINT_IP=$(docker container inspect alerting-endpoint | jq -r --arg network ${NETWORK} '.[].NetworkSettings.Networks[$network].IPAddress')
+echo "Done spinning up alerting interface!"
+
+readarray -t FULL_INTEGRATION_TEST_SUITE < <(yq e -r -o=j '.jobs.[].strategy.matrix | select(.integration-test-arg) | .integration-test-arg[]' .github/workflows/.reusable-integration-test.yml)
+
+FAILED_INTEGRATION_TESTS=()
+SUCCESSFUL_INTEGRATION_TESTS=()
+SCRIPT_FAILURES=()
+
+INTEGRATION_TEST_SUITE=( "${INTEGRATION_TESTS_TO_RUN[@]:-${FULL_INTEGRATION_TEST_SUITE[@]}}" )
+
+if [[ ${EXTENDED} == "true" || "${INTEGRATION_TESTS_TO_RUN[@]}" =~ "self-hosted-notary" ]]; then
+  docker pull docker.io/securesystemsengineering/testimage:self-hosted-notary-signed
+  PREFIXED_DIGEST=$(docker images --digests | grep self-hosted-notary-signed | awk '{print $3}')
+  export DIGEST=$(echo ${PREFIXED_DIGEST#sha256:})
+  docker run -d --name notary-signer -p 7899:7899 -v ./tests/data/notary_service_container/signer:/etc/docker/notary-signer/ --network ${NETWORK} notary:signer -config=/etc/docker/notary-signer/config.json
+  NOTARY_SIGNER_IP=$(docker container inspect notary-signer | jq -r --arg network ${NETWORK} '.[].NetworkSettings.Networks[$network].IPAddress')
+  docker run -d --name notary-server -p 4443:4443 --add-host notary.signer:${NOTARY_SIGNER_IP} -v ./tests/data/notary_service_container/server:/etc/docker/notary-server --network ${NETWORK} notary:server -config=/etc/docker/notary-server/config.json -logf=json
+  export NOTARY_SERVER_IP=$(docker container inspect notary-server | jq -r --arg network ${NETWORK} '.[].NetworkSettings.Networks[$network].IPAddress')
+  docker build --build-arg "DIGEST=${DIGEST}" -f tests/integration/Dockerfile.populate_notary . -t populate-notary
+  docker run --network ${NETWORK} --add-host notary.server:${NOTARY_SERVER_IP} populate-notary
+  export NOTARY_IP=${NOTARY_SERVER_IP}
+else
+  INTEGRATION_TESTS_TO_SKIP+=("self-hosted-notary")
+fi
+
+for INTEGRATION_TEST in ${INTEGRATION_TEST_SUITE[@]}; do
+  if [[ "${INTEGRATION_TESTS_TO_SKIP[@]}" =~ "${INTEGRATION_TEST}" ]]; then
+    echo "Skipping integration test ${INTEGRATION_TEST}"
+  else
+    echo "Running integration test ${INTEGRATION_TEST}..."
+    mkdir ./tmp/${INTEGRATION_TEST}
+    IS_INTEGRATION_TEST_RUNNING="true" #this will be wrong when there is a failure with tee because in that case the integration test won't be started
+    ./tests/integration/integration-test.sh ${INTEGRATION_TEST} | tee ./tmp/${INTEGRATION_TEST}/test.log || true
```

Comment:
```suggestion
    ./tests/integration/integration-test.sh ${INTEGRATION_TEST} &| tee ./tmp/${INTEGRATION_TEST}/test.log || true
```
?

Reply:
What does it do?

Reply:
Pipe stderr and stdin through the pipe instead of just stdin, which would catch any missing 2>&1 in the other script

Reply:
Oh, oops, I thought thats `|&`  😁

But yep, I think it makes sense 👍